sdk/storage/Azure.Storage.DataMovement.Files.Shares/src/DataMovementSharesExtensions.cs

```diff
@@ -264,6 +267,18 @@ internal static partial class DataMovementSharesExtensions
             {
                 existingProperties.RawProperties.WriteKeyValue(DataMovementConstants.ResourceProperties.CacheControl, fileProperties.CacheControl);
             }
+            if (existingProperties.LastModifiedTime == default)
+            {
+                existingProperties.LastModifiedTime = fileProperties.LastModified;
+            }
+            if (existingProperties.ETag == default)
+            {
+                existingProperties.ETag = fileProperties.ETag;
+            }
+            if (existingProperties.ResourceLength == default)
+            {
+                existingProperties.ResourceLength = fileProperties.ContentLength;
```

Comment:
Made ResourceLength settable becuase of this change here.

There's a case where when we do directory transfers, the ShareFileItemProperties that comes back during enumeration does not have all the information we need. And then when we do, do a regular GetProperties call, this is where we ensure that information is updated. That way we do store the ResourceLength when we do get one.